3p/recaptcha.js

```diff
@@ -41,28 +47,110 @@ const TAG = 'RECAPTCHA';
 /** @const {string} */
 const RECAPTCHA_API_URL = 'https://www.google.com/recaptcha/api.js?render=';
 
+/** {?IframeMessaginClient} **/
+let iframeMessagingClient = null;
+
+/**
+ * Initialize 3p frame.
+ */
+function init() {
+  initLogConstructor();
+  setReportError(console.error.bind(console));
+}
+
+// Immediately call init
+init();
+
 /**
- * @param {!Window} global
- * @param {!Object} data
+ * Main function called by the recaptcha bootstrap frame
  */
-export function recaptcha(global, data) {
+function recaptcha() {
 
-  validateData(data, ['sitekey', 'action']);
+  /**
+   *  Get the data from our name attribute
+   * sitekey {string} - reCAPTCHA sitekey used to identify the site
+   * sentinel {string} - string used to psuedo-confirm that we are
+   *    receiving messages from the recaptcha frame
+   */
+  let dataObject;
+  try {
+    dataObject = parseJson(window.name);
+  } catch (e) {
+    dev().error(TAG + ' Could not parse the window name.');
+    return;
+  }
 
-  const {sitekey} = data;
+  // Get our sitekey from the iframe name attribute
+  dev().assert(
+      dataObject.sitekey,
+      'The sitekey is required for the <amp-recaptcha-input> iframe %s',
```

Comment:
curious what will be printed out given dataObject is an object.

Reply:
Nice catch It outputted: `Uncaught Error: The sitekey is required for the <amp-recaptcha-input> iframe [object Object]`.

Reply:
@torch2424 Actually I think we don't see a good reason to append the dataObject to the error message. If `siteKey` is missing, then the only left was `sentinel=amp-recaptcha`. Do you think we can delete it?